eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,116 @@
+$Language = "go"
+$packagePattern = "go.mod"
+# rewrite from ChangeLog-Operations.ps1 used in Get-ChangeLogEntriesFromContent for go uses vx.x.x as version number
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+# constants for go version fetch
+$GO_VERSION_REGEX = ".+\s*=\s*`".*v?(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
+$VERSION_FILE_SUFFIXS = "*constants.go", "*version.go"
+
+# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
+function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+{
+    $workFolder = $pkg.Directory
+    $releaseNotes = ""
+    $namespaceName = $workFolder.Name
+    $rpName = $workFolder.Parent.Name
+
+    $pkgId = "sdk/$rpName/$namespaceName"
+    $pkgVersion = Get-Version $workFolder
+
+    $changeLogLoc = @(Get-ChildItem -Path $workFolder -Recurse -Include "CHANGELOG.md")[0]
+    if ($changeLogLoc)
+    {
+        $releaseNotes = Get-ChangeLogEntryAsString -ChangeLogLocation $changeLogLoc -VersionString v$pkgVersion
+    }
+
+    $resultObj = New-Object PSObject -Property @{
+        PackageId      = $pkgId
+        PackageVersion = $pkgVersion
+        ReleaseTag     = "$pkgId/v$pkgVersion"
+        Deployable     = $true
+        ReleaseNotes   = $releaseNotes
+    }
+
+    return $resultObj
+}
+
+# get version from specific files (*constants.go, *version.go)
+function Get-Version ($pkgPath)
+{
+    # find any file with surfix
+    $versionFiles = [Collections.Generic.List[String]]@()
+    foreach ($versionFileSuffix in $VERSION_FILE_SUFFIXS)
+    {
+        Get-ChildItem -Recurse -Path $pkgPath -Filter $versionFileSuffix | ForEach-Object {
+            Write-Host "Adding $_ to list of version files"
+            $versionFiles.Add($_)
+        }
+    }
+    
+    # for each version file, use regex to search go version num
+    foreach ($versionFile in $versionFiles)
+    {
+        $content = Get-Content $versionFile
+        $content = $content.Split("`n")
+        try
+        {
+            # walk the document, finding where the version number are
+            foreach ($line in $content)
+            {
+                if ($line -match $GO_VERSION_REGEX)
+                {
+                    return $matches["version"]
+                }
+            }
+        }
+        catch
+        {
+            Write-Error "Error parsing version."
+            Write-Error $_
+        }
+    }
+
+    Write-Host "Cannot find release version."
+    exit 1
+}
+
+# rewrite from artifact-metadata-parsing.ps1 as all 0.x.x version should be considered prerelease in go sdk
```

Comment:
FYI I'm adding support for v0 versions in our core SemVer helper with https://github.com/Azure/azure-sdk-tools/pull/1972. If that is included do we still need to duplicate this entire function? Hopefully not.

Reply:
It will be great that semver helper can handle this. I'll remove it after your pr merged and automatic sync done.

Reply:
My SemVer change is in.